src/main/java/com/trainer/name/service/TrainerService.java

```diff
@@ -52,4 +52,22 @@ public Trainer findById(int id) throws TrainerNotFoundException {
         return trainerMapper.findById(id).orElseThrow(() -> new TrainerNotFoundException("idが" + id + "のトレーナーはいません"));
     }
 
+    private boolean isEmailUnique(String email) {
+        int count = trainerMapper.countByEmail(email);
+        return count == 0;
+    }
+
+    public Trainer insert(String name, String email) {
+        if (name == null || email == null || name.trim().isEmpty() || email.trim().isEmpty()) {
+            throw new IllegalArgumentException("名前とメールアドレスは必須です");
+        }
+        if (!isEmailUnique(email)) {
+            throw new IllegalArgumentException("このメールアドレスは既に使用されています");
+        }
```

Comment:
`DuplicateEmailException`を使った方が良いですね。
```suggestion
        if (!isEmailUnique(email)) {
            throw new DuplicateEmailException("このメールアドレスは既に使用されています");
        }
```

あと、名前が重複していた場合のハンドリングも実装しないといけないと思ったのですが、いかがでしょう？
`DuplicateNameException`を使って実装する必要があると思います。

Reply:
名前が重複していた際の処理を実装しました
75dd305bd0f9316d402fba350d27944de3706fbe
![image](https://github.com/Ito-0131/10th-Assignment/assets/138368847/d2df93c0-1962-4236-a4b3-0e0028965b80)
![image](https://github.com/Ito-0131/10th-Assignment/assets/138368847/5b2c046d-0650-4874-abde-2a8ffb11b4a4)